pkg/helm/handlers/handler_test.go

```diff
@@ -290,7 +330,202 @@ func TestHelmHandlers_HandleGetChart(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedResponse    string
+		fakeResp            map[string]interface{}
+		expectedContentType string
+		error
+		httpStatusCode int
+	}{
+		{
+			name:                "Invalid chart rollback release test",
+			error:               errors.New("Chart path is invalid"),
+			expectedResponse:    `{"error":"Failed to rollback helm releases: Chart path is invalid"}`,
+			httpStatusCode:      http.StatusBadGateway,
+			expectedContentType: "application/json",
+		},
+		{
+			name:             "Valid chart rollback release test",
+			expectedResponse: `{"status":"rollback successful"}`,
+			fakeResp: map[string]interface{}{
+				"status": "rollback successful",
+			},
+			expectedContentType: "application/json",
+			error:               nil,
+			httpStatusCode:      http.StatusOK,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handlers := fakeHelmHandler()
+			var request *http.Request
+
+			handlers.rollbackRelease = fakeRollbackRelease(tt.fakeResp, tt.error)
+
+			request = httptest.NewRequest("", "/foo?revision=1", strings.NewReader(`{"name": "test", "namespace":"test"}`))
+			response := httptest.NewRecorder()
+
+			handlers.HandleRollbackRelease(&auth.User{}, response, request)
+
+			if response.Code != tt.httpStatusCode {
+				t.Errorf("response code should be %v but got %v", tt.httpStatusCode, response.Code)
+			}
+			if response.Header().Get("Content-Type") != tt.expectedContentType {
+				t.Errorf("content type should be %s but got %s", tt.expectedContentType, response.Header().Get("Content-Type"))
+			}
+			if response.Body.String() != tt.expectedResponse {
+				t.Errorf("response body not matching expected is %s and received is %s", tt.expectedResponse, response.Body.String())
+			}
+		})
+	}
+}
 
+func TestHelmHandlers_HandleHelmUpgradeRelease(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedResponse    string
+		expectedContentType string
+		release             *release.Release
+		error
+		httpStatusCode int
+	}{
+		{
+			name:                "Invalid chart path upgrade release test",
+			error:               errors.New("Chart path is invalid"),
+			expectedResponse:    `{"error":"Failed to upgrade helm release: Chart path is invalid"}`,
+			httpStatusCode:      http.StatusBadGateway,
+			expectedContentType: "application/json",
+		},
+		{
+			name:                "Valid chart upgrade release",
+			expectedResponse:    `{"name":"Test"}`,
+			release:             &fakeRelease,
+			expectedContentType: "application/json",
+			error:               nil,
+			httpStatusCode:      http.StatusOK,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handlers := fakeHelmHandler()
+			var request *http.Request
+
+			handlers.upgradeRelease = fakeUpgradeRelease(tt.release, tt.error)
+
+			request = httptest.NewRequest("", "/foo", strings.NewReader("{}"))
```

Comment:
we should provide some payload and check if it is decoded and used properly.

Reply:
Added test to validate this

Reply:
Where is that test?